url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/util/bitfield/BitField.java

```diff
@@ -0,0 +1,286 @@
+package net.hetimatan.util.bitfield;
+
+import java.util.Random;
+
+import net.hetimatan.util.url.PercentEncoder;
+
+
+public class BitField {
+	public static final int[] BIT = {0xFF, 0x80, 0xC0, 0xE0, 0xF0, 0xF8, 0xFC, 0xFE};
+
+	private byte[] mBitfield = new byte[0];
+	private int mBitsize = 0;
+	private Random mR = null;
+	private int[] mShuffleList= new int[]{0,1,2,3,4,5,6,7};
+
+	
+	public static BitField relative(BitField ina, BitField inb, BitField out) {
+		if(out == null) {
+			int len = ina.lengthPerBit();
+			out = new BitField(len);
+		}
+		int len = out.lengthPerByte();
+		if(len>inb.lengthPerByte()) {
+			len = inb.lengthPerByte();
+		}
+		for(int i=0;i<out.lengthPerByte();i++) {
+			out.mBitfield[i] = (byte)(0xFF&out.mBitfield[i]);
+		}
+		for(int i=0;i<len;i++) {
+			out.mBitfield[i] = (byte)(0xFF&ina.mBitfield[i]&(~inb.mBitfield[i]));
+		}
+		out.update();
+		return out;
+	}
+
+	public static int numOfOn(BitField field) {
+		int ret = 0;
+		int size = field.lengthPerBit();
+		for(int i=0;i<size;i++) {
+			
+		}
+		return ret;
+	}
+
+	public void update(){}
+	public BitField(int bitsize) {
+		mR = new Random(System.currentTimeMillis());
+		mBitsize = bitsize;
+		int byteSize = bitsize/8;
+		if((bitsize%8)!=0) {
+			byteSize+=1;
+		}
+		mBitfield = new byte[byteSize];
+		oneClear();
+	}
+
+	//
+	// TODO next work following method is wrong
+	//
+	public int getOffPieceAtRandom() {
+		return getPieceAtRandom(true);
+	}
+
+	public int getOnPieceAtRandom() {
+		return getPieceAtRandom(false);
+	}
+
+	//
+	// TODO next work following method is wrong
+	//
+	public int getPieceAtRandom(boolean isOff) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.